scale/job/configuration/json/job/job_config.py

```diff
@@ -0,0 +1,212 @@
+
+        self._populate_default_values()
+        self._validate_mounts()
+        self._validate_settings()
+
+    def get_dict(self):
+        """Returns the internal dictionary representing this configuration
+
+        :returns: The internal dictionary
+        :rtype: dict
+        """
+
+        return self._configuration
+
+    def get_mount_volume(self, mount_name, volume_name, container_path, mode):
+        """Returns the volume that has been configured for the given mount name. If the given mount is not defined in
+        this configuration, None is returned.
+
+        :param mount_name: The name of the mount defined in the job type
+        :type mount_name: string
+        :param volume_name: The name of the volume
+        :type volume_name: string
+        :param container_path: The path within the container onto which the volume will be mounted
+        :type container_path: string
+        :param mode: Either 'ro' for read-only or 'rw' for read-write
+        :type mode: string
+        :returns: The volume that should be mounted into the job container, possibly None
+        :rtype: :class:`job.configuration.volume.Volume`
+        """
+
+        if mount_name not in self._configuration['mounts']:
+            return None
+
+        volume = None
+        mount_config = self._configuration['mounts'][mount_name]
+        mount_type = mount_config['type']
+        if mount_type == 'host':
+            host_path = mount_config['host_path']
+            volume = Volume(container_path, mode, is_host=True, host_path=host_path)
+        elif mount_type == 'volume':
+            driver = mount_config['driver']
+            driver_opts = mount_config['driver_opts']
+            volume = Volume(container_path, mode, is_host=False, name=volume_name, driver=driver,
+                            driver_opts=driver_opts)
+
+        return volume
+
+    def get_setting_value(self, name):
+        """Returns the value of the given setting if defined in this configuration, otherwise returns None
+
+        :param name: The name of the setting
+        :type name: string
+        :returns: The value of the setting, possibly None
+        :rtype: string
+        """
+
+        if name in self._configuration['settings']:
+            return self._configuration['settings'][name]
+
+        return None
+
+    def _convert_configuration(self):
+        """Converts the configuration from a previous schema version
+
+        :raises :class:`job.configuration.exceptions.InvalidJobConfiguration`: If the given configuration is invalid
+        """
+
+        # Validate/process the dict according to the previous version
+        self._configuration = previous_interface.JobConfiguration(self._configuration).get_dict()
+
+        self._configuration['version'] = SCHEMA_VERSION
+        self._configuration['settings'] = self._configuration['default_settings']
+        del self._configuration['default_settings']
+
+    def _populate_default_values(self):
```

Comment:
Same comment as with execution config with regards to defaults being stored in the JSON_SCHEMA block above.